static/src/javascripts/projects/commercial/modules/liveblog-adverts.js

```diff
@@ -94,11 +100,11 @@ const insertAds = (paras) => {
 
 const fill = (rules) =>
 	spaceFiller.fillSpace(rules, insertAds).then((result) => {
-		// Before the refactor of fillSpace in https://github.com/guardian/frontend/pull/24599,
-		// since insertAds returned void, result is only ever undefined and the code path below was dead.
-		// TODO investigate the impact of removing the following line.
-		result = undefined;
-		if (result && AD_COUNTER < MAX_ADS) {
```

Comment:
@simonbyford and I noticed that spacefinder [throws an error](https://github.com/guardian/frontend/blob/73031cb6ba235d6696f6a00d79730e57dd817023/static/src/javascripts/projects/common/modules/spacefinder.js#L316) when no candidates are eligible for an ad slot, so that [spacefiller's return value](https://github.com/guardian/frontend/blob/97b449adf3cc5f31806e988e26a49f9b7a8ff82a/static/src/javascripts/projects/common/modules/article/space-filler.ts#L72) and `result` are `false`.
This means that at the moment if an execution of spacefinder fails for any reason, including if it doesn't find any candidates, we stop listening for more blocks to be inserted.
We think it would be more appropriate to ignore the return value of `fillSpace` and always listen for the insertion of new blocks, regardless of the outcome of previous runs of spacefinder.